app/preview.py

```diff
@@ -139,6 +151,32 @@ def view_letter_template(filetype):
         raise e
 
 
+@preview_blueprint.route("/precompiled-preview.png", methods=['POST'])
+@auth.login_required
+@statsd(namespace="template_preview")
+def view_precompiled_letter():
+    try:
+        encoded_string = request.get_data()
+
+        if not encoded_string:
+            abort(400)
+
+        pdf = base64.decodestring(encoded_string)
+
+        return send_file(**png_from_pdf(
+            pdf, page_number=int(request.args.get('page', 1)), hide_notify=True
+        ))
+
+    # catch invalid pdfs
+    except MissingDelegateError as e:
+        current_app.logger.error(str(e))
```

Comment:
Add some context here? What fails when there is a a MissingDelegateError?
 e.g. current_app.logger.error("Failed to decode base64String" + str(e))